forui/lib/src/widgets/select_group/select_group_item.dart

```diff
@@ -1,136 +1,270 @@
+  @override
+  bool updateShouldNotify(FSelectGroupItemData old) =>
+      controller != old.controller || style != old.style || selected != old.selected;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('controller', controller))
+      ..add(DiagnosticsProperty('style', style))
+      ..add(FlagProperty('selected', value: selected, ifTrue: 'selected'));
+  }
+}
+
+/// A [FSelectGroupItem]'s state.
+typedef FSelectGroupItemState<T> = ({
+  FSelectGroupController<T> controller,
+  FSelectGroupStyle style,
+  T value,
+  bool selected
+});
+
+/// Represents a selection in a [FSelectGroup].
+abstract class FSelectGroupItem<T> extends StatelessWidget {
+  /// The item's value.
+  final T value;
+
+  /// Creates a [FSelectGroupItem].
+  const factory FSelectGroupItem({
+    required T value,
+    required ValueWidgetBuilder<FSelectGroupItemState<T>> builder,
+    Widget? child,
+    Key? key,
+  }) = _Builder;
+
+  /// Creates a checkbox wrapped in a [FSelectGroupItem].
+  // TODO: Making this const causes a false positive in the list_element_type_not_assignable lint when declared
+  //       in a const list inside FSelectGroup. I still can't replicate this issue in other cases.
+  // ignore: prefer_const_constructors_in_immutables
   factory FSelectGroupItem.checkbox({
     required T value,
     FCheckboxSelectGroupStyle? style,
     Widget? label,
     Widget? description,
-    String? semanticLabel,
     Widget? error,
-    bool enabled = true,
-    bool autofocus = false,
+    String? semanticLabel,
+    bool enabled,
+    bool autofocus,
     FocusNode? focusNode,
     ValueChanged<bool>? onFocusChange,
     Key? key,
-  }) =>
-      FSelectGroupItem(
-        value: value,
-        builder: (context, onChange, selected) {
-          final computedStyle = style ?? context.theme.selectGroupStyle.checkboxStyle;
-
-          return Padding(
-            padding: computedStyle.padding,
-            child: FCheckbox(
-              style: computedStyle,
-              label: label,
-              description: description,
-              semanticLabel: semanticLabel,
-              error: error,
-              value: selected,
-              onChange: (state) => onChange(value, state),
-              enabled: enabled,
-              autofocus: autofocus,
-              focusNode: focusNode,
-              onFocusChange: onFocusChange,
-              key: key,
-            ),
-          );
-        },
-      );
-
-  /// Creates a [FSelectGroupItem] that wraps a [FRadio].
+  }) = _Checkbox;
+
+  /// Creates a radio button wrapped in a [FSelectGroupItem].
+  // TODO: Making this const causes a false positive in the list_element_type_not_assignable lint when declared
+  //       in a const list inside FSelectGroup. I still can't replicate this issue in other cases.
+  // ignore: prefer_const_constructors_in_immutables
```

Comment:
I'm not too familiar with the issue here, but would it make sense to track this with an issue?